docs/endpoints.md

```diff
@@ -42,3 +44,76 @@ class User(HTTPEndpoint):
 
 HTTP endpoint classes will respond with "405 Method not allowed" responses for any
 request methods which do not map to a corresponding handler.
+
+### WebSocketEndpoint
+
+The `WebSocketEndpoint` class is an ASGI application that presents a wrapper around
+the functionality of a `WebSocket` instance. 
+
+The ASGI connection details are accessible on the endpoint instance:
+
+* `.scope` - The request scope.
+* `.websocket` - The `WebSocket` instance.
+* `.close_code` - The close code used when the websocket session is closed.
+* `.kwargs` - The `scope` kwargs.
```

Comment:
Sorry, change of heart. I reckon we drop all of these except `.scope` in order to match `HTTPEndpoint` more neatly.